mockative-processor/src/main/kotlin/io/mockative/kotlinpoet/ClassName.Mockative.kt

```diff
@@ -27,35 +27,35 @@ private fun mockValue(type: ClassName): CodeBlock {
 }
 
-internal fun valueOf(type: ClassName): CodeBlock {
+internal fun valueOf(type: ClassName): Pair<Boolean, CodeBlock> {
     return when (type) {
-        BOOLEAN -> CodeBlock.of("%L", false)
-        BYTE -> CodeBlock.of("%L", 0)
-        SHORT -> CodeBlock.of("%L", 0)
-        CHAR -> CodeBlock.of("%L.toChar()", 0)
-        INT -> CodeBlock.of("%L", 0)
-        LONG -> CodeBlock.of("%L", 0)
-        FLOAT -> CodeBlock.of("%L", 0)
-        DOUBLE -> CodeBlock.of("%L", 0)
-
-        BOOLEAN_ARRAY -> CodeBlock.of("%T(0)", BOOLEAN_ARRAY)
-        BYTE_ARRAY -> CodeBlock.of("%T(0)", BYTE_ARRAY)
-        SHORT_ARRAY -> CodeBlock.of("%T(0)", SHORT_ARRAY)
-        CHAR_ARRAY -> CodeBlock.of("%T(0)", CHAR_ARRAY)
-        INT_ARRAY -> CodeBlock.of("%T(0)", INT_ARRAY)
-        LONG_ARRAY -> CodeBlock.of("%T(0)", LONG_ARRAY)
-        FLOAT_ARRAY -> CodeBlock.of("%T(0)", FLOAT_ARRAY)
-        DOUBLE_ARRAY -> CodeBlock.of("%T(0)", DOUBLE_ARRAY)
-        STRING -> CodeBlock.of("\"\"")
-        CHAR_SEQUENCE -> CodeBlock.of("\"\"")
-
-        ARRAY_LIST -> CodeBlock.of("%T()", ARRAY_LIST)
-        ARRAY_DEQUE -> CodeBlock.of("%T()", ARRAY_DEQUE)
-        LINKED_HASH_MAP -> CodeBlock.of("%T()", LINKED_HASH_MAP)
-        HASH_MAP -> CodeBlock.of("%T()", HASH_MAP)
-        LINKED_HASH_SET -> CodeBlock.of("%T()", LINKED_HASH_SET)
-        HASH_SET -> CodeBlock.of("%T()", HASH_SET)
-
-        else -> mockValue(type)
+        BOOLEAN -> Pair(false, CodeBlock.of("%L", false))
+        BYTE -> Pair(false, CodeBlock.of("%L", 0))
+        SHORT -> Pair(false, CodeBlock.of("%L", 0))
+        CHAR -> Pair(false, CodeBlock.of("%L.toChar()", 0))
+        INT -> Pair(false, CodeBlock.of("%L", 0))
+        LONG -> Pair(false, CodeBlock.of("%L", 0))
+        FLOAT -> Pair(false, CodeBlock.of("%L", 0))
+        DOUBLE -> Pair(false, CodeBlock.of("%L", 0))
+
+        BOOLEAN_ARRAY -> Pair(false, CodeBlock.of("%T(0)", BOOLEAN_ARRAY))
+        BYTE_ARRAY -> Pair(false, CodeBlock.of("%T(0)", BYTE_ARRAY))
+        SHORT_ARRAY -> Pair(false, CodeBlock.of("%T(0)", SHORT_ARRAY))
+        CHAR_ARRAY -> Pair(false, CodeBlock.of("%T(0)", CHAR_ARRAY))
+        INT_ARRAY -> Pair(false, CodeBlock.of("%T(0)", INT_ARRAY))
+        LONG_ARRAY -> Pair(false, CodeBlock.of("%T(0)", LONG_ARRAY))
+        FLOAT_ARRAY -> Pair(false, CodeBlock.of("%T(0)", FLOAT_ARRAY))
+        DOUBLE_ARRAY -> Pair(false, CodeBlock.of("%T(0)", DOUBLE_ARRAY))
+        STRING -> Pair(false, CodeBlock.of("\"\""))
+        CHAR_SEQUENCE -> Pair(false, CodeBlock.of("\"\""))
+
+        ARRAY_LIST -> Pair(false, CodeBlock.of("%T()", ARRAY_LIST))
+        ARRAY_DEQUE -> Pair(false, CodeBlock.of("%T()", ARRAY_DEQUE))
+        LINKED_HASH_MAP -> Pair(false, CodeBlock.of("%T()", LINKED_HASH_MAP))
+        HASH_MAP -> Pair(false, CodeBlock.of("%T()", HASH_MAP))
+        LINKED_HASH_SET -> Pair(false, CodeBlock.of("%T()", LINKED_HASH_SET))
+        HASH_SET -> Pair(false, CodeBlock.of("%T()", HASH_SET))
```

Comment:
Either add a function that returns a pair of fale to a provided codeblock like:
```kotlin
fun funName(isMock: Boolean = false, codeBlock: CodeBlock): Pair<Boolean, CodeBlock> = Pair(isMock, codeBlock)
```
And then call this, as having so many `false` statements is very error-prone. This also increases readability while also giving a name to the `Boolean`.

Reply:
I refactored this to use a data class. I hope I understood you correctly.